rskj-core/src/test/java/co/rsk/peg/BridgeTest.java

```diff
@@ -2266,6 +2266,40 @@ void getProposedFederationAddress(MessageCall.MsgType msgType, ActivationConfig
         }
     }
 
+    @ParameterizedTest()
+    @MethodSource("msgTypesAndActivations")
+    void getProposedFederationCreationTime(MessageCall.MsgType msgType, ActivationConfig activationConfig) throws VMException {
+        Transaction rskTxMock = mock(Transaction.class);
+        doReturn(true).when(rskTxMock).isLocalCallTransaction();
+
+        BridgeSupport bridgeSupportMock = mock(BridgeSupport.class);
+        Instant expectedCreationTime = Instant.EPOCH;
+        when(bridgeSupportMock.getProposedFederationCreationTime()).thenReturn(Optional.of(expectedCreationTime));
+
+        Bridge bridge = bridgeBuilder
+            .transaction(rskTxMock)
+            .activationConfig(activationConfig)
+            .bridgeSupport(bridgeSupportMock)
+            .msgType(msgType)
+            .build();
+
+        CallTransaction.Function function = BridgeMethods.GET_PROPOSED_FEDERATION_CREATION_TIME.getFunction();
+        byte[] data = function.encode();
+
+        if (activationConfig.isActive(ConsensusRule.RSKIP419, 0)) {
```

Comment:
This test is hard to follow, having this multi-level `if`. I believe this should be extracted into separate test methods, eliminating the conditional for the assertions. Wdyt?

Reply:
I think we had this same discussion for every PR that add a test here 😬

All these test will be later refactored